mod_reforged/hooks/skills/effects/smoke_effect.nut

```diff
@@ -0,0 +1,29 @@
+::mods_hookExactClass("skills/effects/smoke_effect", function (o) {
+    o.m.RangedDefenseBonus = 30;
+
+    local getTooltip = o.getTooltip;
+	o.getTooltip = function()
+	{
+        local ret = getTooltip();
+        foreach (entry in ret)
+        {
+            if (entry.id == 12 && entry.icon == "ui/icons/ranged_defense.png")
+            {
+                // We replace a single line of the vanilal tooltip without touching the rest
+                entry.text = ::MSU.Text.colorGreen("+" + this.m.RangedDefenseBonus) + " Ranged Defense";
+                break;
+            }
+        }
+        return ret;
+	}
+
+    local onUpdate = o.onUpdate;
+    o.onUpdate = function( _properties )
+    {
+        onUpdate(_properties);
+        if (this.m.IsGarbage == true) return;     // Simple way of checking whether the line above skipped the buffs because the smoke effect is not active anymore
+
+        _properties.RangedDefenseMult *= 0.5;   // Hard-Coded reversing of the vanilla buffs. They multiply it with 2.0
```

Comment:
Instead of this reversal method, perhaps overwrite it completely? What if a mod hooks it to change `RangedSkillMult` by 3.0 during this function. Then we are not fully reversing it anyway - so I don't think we have to care about mod compatibility here as we are completely changing how this effect behaves.

Reply:
But replacing that function doesn't fix the issue you are describing if that mod loads after our mod. Then their adjustments still take place and warp the effects of smoke bomb to an undesired level.

Going all-in into hooking here at least provides compatibility with most other adjustments that a modder or vanilla could do to the smoke-bomb effect

Reply:
I replaced that line with a better solution now. 
I don't reverse the effects but instead just reset that property to the value it was before the vanilla function was called.

Reply:
I think overwriting the function is better in this case than resetting the property like that. Because if you want to be incompatible with a mod's changes, better make it explicit. Otherwise this can lead to partial changes being overwritten and partial not - which is hard to track down in the code's logic and catch with multiple mods. @Enduriel please opinionate.

Reply:
Based on our discussion in Discord, I am ok with the complete reverting of changes to RangedDefenseMult in our hook on onUpdate.